contracts/command-batcher/src/contract.rs

```diff
@@ -1,44 +1,350 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cosmwasm_std::{
+    from_binary, to_binary, wasm_execute, Binary, BlockInfo, Deps, DepsMut, Env, HexBinary,
+    MessageInfo, QuerierWrapper, QueryRequest, Reply, Response, StdError, StdResult, SubMsg,
+    WasmMsg, WasmQuery,
+};
+use cw_utils::{parse_reply_execute_data, MsgExecuteContractResponse};
+
+use std::collections::HashMap;
+
+use axelar_wasm_std::{Participant, Snapshot};
+use connection_router::msg::Message;
+use multisig::{msg::Multisig, types::MultisigState};
+use service_registry::state::Worker;
 
 use crate::{
     error::ContractError,
+    events::Event,
     msg::ExecuteMsg,
-    msg::{GetProofResponse, QueryMsg},
+    msg::{GetProofResponse, InstantiateMsg, ProofStatus, QueryMsg},
+    state::{
+        Config, COMMANDS_BATCH, CONFIG, PROOF_BATCH_MULTISIG, REPLY_ID_COUNTER, REPLY_ID_TO_BATCH,
+    },
+    types::{BatchID, CommandBatch, ProofID},
 };
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn execute(
-    _deps: DepsMut,
+pub fn instantiate(
+    deps: DepsMut,
     _env: Env,
-    _info: MessageInfo,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let admin = deps.api.addr_validate(info.sender.as_str())?;
```

Comment:
admin should be a parameter in the message, so the sender can specify a different admin than themselves